frontend/projects/upgrade/src/app/shared-standalone-component-lib/components/common-section-card-content/common-section-card-content.component.ts

```diff
@@ -17,15 +17,19 @@ import { SharedModule } from '../../../shared/shared.module';
 //
 // Example Usage:
 //
-// contentDetails = [
-//   { key: 'name' },
-//   { description: 'something' },
-//   { tags: ['Tag1', 'Tag2'] },
-//   { Appcontext: 'Context1' },
-// ];
+// contentDetails = {
+//   Key: 'name',
+//   Description: 'something',
+//   Tags: ['Tag1', 'Tag2'],
+//   app_Context: 'Context1',
+// };
 //
 // Simply pass the data to the component as shown below:
 // <app-common-section-card-content [data]="contentDetails"></app-common-section-card-content>
 export class CommonSectionCardContentComponent {
-  @Input() data: { key: string; value: string }[] = [];
+  @Input() data!: KeyValueFormat;
```

Comment:
I think we want something more specific than a general "KeyValueFormat", because it's important that Tags is an array.

Can we try this as an interface that is specific to these overview components?

```
export interface OverviewDetails {
  [key: string]: string;
  ['Tags']?: string[];
}
```

Reply:
Giving this error Property '['Tags']' of type 'string[]' is not assignable to 'string' index type 'string'.
can't we keep it this way?
```
export interface KeyValueFormat {
  [key: string]: string | string[];
}
```

Reply:
please try this instead. We're describing a specific type for the needs of this specific component only, not a general format that needs to be on the `types` file.

export type OverviewDetails = {
  [key: string]: string;
} & {
  Tags?: string[];
};

Reply:
this is done, little diff but done